packages/hash/api/src/graphql/opentelemetry.ts

```diff
@@ -0,0 +1,65 @@
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-grpc";
+import { registerInstrumentations } from "@opentelemetry/instrumentation";
+import {
+  ExpressInstrumentation,
+  ExpressLayerType,
+} from "@opentelemetry/instrumentation-express";
+import { GraphQLInstrumentation } from "@opentelemetry/instrumentation-graphql";
+import { HttpInstrumentation } from "@opentelemetry/instrumentation-http";
+import { Resource } from "@opentelemetry/resources";
+import { SimpleSpanProcessor } from "@opentelemetry/sdk-trace-base";
+import { NodeTracerProvider } from "@opentelemetry/sdk-trace-node";
+
+import { logger } from "../logger";
+
+const traceTimeout = 5000;
+
+const unregisterInstrumentations = registerInstrumentations({
+  instrumentations: [
+    new HttpInstrumentation(),
+    new ExpressInstrumentation({
+      ignoreLayersType: [ExpressLayerType.MIDDLEWARE],
+    }),
+    new GraphQLInstrumentation({
+      allowValues: true,
+      depth: 5,
+    }),
+  ],
+});
+
+export const registerOpenTelemetryTracing = (
+  otlpGrpcEndpoint: string | null,
+): (() => void) => {
+  if (!otlpGrpcEndpoint) {
+    logger.info(
+      "No OpenTelemetry Protocol endpoint given. Not sending tracespans anywhere.",
+    );
```

Comment:
Do we perhaps want to set this as a `warn`?

Reply:
I did consider that, I think I will do that. The app will function fine without tracing, though, but of course it could be useful to have it be a `warn` when it's misconfigured.